src/rrt/Tree.hpp

```diff
@@ -309,7 +350,21 @@ class Tree {
         }
 
         // Add a node to the tree for this state
+        int lengthT = sizeof(intermediateState) / sizeof(0.0);
+        double* data = new double[lengthT];
+        if (NULL == _TToArray) {
+            for (int i = 0; i < lengthT; i++) {
+                data[i] = intermediateState[i];
+            }
+        } else {
+            data = _TToArray(intermediateState);
```

Comment:
Sorry, that wasn't really clear on my part. You can keep both `TToArray` and `ArrayToT`, but I would almost pass the array as an argument or use some sort of c++ container object instead of a raw array (`std::vector`, `std::array` etc).

The first gives you control to do both the memory allocation for the array as well as the deallocation. You already have it allocated so you might as well pass it in. For this case, `TToArray` will be `void TToArray(double* array, T state)`

The second allows for bounds checking when accessing the data as well as the overall size of the data. This can help solve that problem where you aren't able to get the size of the data from the `intermediateState` object.